### DISCUSSION TITLE
Not getting the image of NFT

### DISCUSSION THREAD

**Discussion by vishnu-geek** [Cyfrin/foundry-full-course-cu#1764] [General]
i deployed the moodnft script minted it succesfully but while importing it to metamask its not showing the actual image
<img width="659" alt="Untitled" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/127754819/c8eb6276-50c9-42dd-ba94-b26b06cf5061">

**Comment by EngrPips** [comment]
Hello @vishnu-geek, I had a similar issue, and I could not resolve it, but someone tried resolving it and wrote a post on how they resolved it. I will share the post link so you can look and see if it will help. here `https://www.linkedin.com/pulse/making-your-nft-collection-yevhen-hrytsenko-mjxif/`. Note: They used  anvil chain, but you could try the same thing for sepolia if you run a sepolia chain.

**Reply by masstrepidation** [reply]
I am also not getting the image, this is json and svg data saved on the blockchain, this should work no matter what, did they change the way the json interacts?

**Reply by EngrPips** [reply]
I mean, I am not sure. You can take a look at this `https://www.linkedin.com/pulse/making-your-nft-collection-yevhen-hrytsenko-mjxif/` and see if it helps.

**Comment by PtSik** [comment]
I also have this issue, the image does show up on the opensea site but in metamask it doesnt

**Reply by EngrPips** [reply]
Same thing on my end, it seems like  a metamask issue.

**Reply by SyedAsadKazmi** [reply]
Yes, **MetaMask** has mentioned about this problem in their [Support Doc](https://support.metamask.io/metamask-portfolio/dashboard/an-nft-is-missing-marked-suspicious-or-not-displaying-correctly-in-metamask-portfolio/#an-nft-is-not-displaying-correctly). And, in that, they've also mentioned that:

> We're working on adding support for more formats and ensuring as many NFTs display correctly as possible.

**Reply by EngrPips** [reply]
Awesome. What a relief to know this.

**Comment by AntoineF4C5** [comment]
I also ran into this issue and the solution for me was to follow these steps:
- Open Metamask
- Go to Settings
- Go to Security and privacy
- Within "Token detection", set "Display NFT Media" to ON (read the warning)
- then go to your NFT, "refresh" it by removing it and adding it again

At least this worked for me, on the anvil chain !

**Reply by EngrPips** [reply]
Awesome, thanks for sharing this with us.
